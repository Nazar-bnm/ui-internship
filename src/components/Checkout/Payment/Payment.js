import React from 'react';
import StripeCheckout from 'react-stripe-checkout';
import { stripeKey } from '../../../config/stripeKey';
import { PAYMENT_SUCCESSED, PAYMENT_FAILED } from '../../../constants/notificationData';
import HttpService from '../../../service/HttpService/httpService';

import './Payment.scss';

const CN = 'payment';

const Payment = (props) => {
  const { totalCount } = props;

  async function handleToken(token) {
    const { showMessage } = props;
    try {
      const response = await new HttpService().post(
        'https://hi3vv.sse.codesandbox.io/checkout',
        { token, totalCount }
      );

      const { status } = response.data;

      if (status === 'success') {
        showMessage(PAYMENT_SUCCESSED);
      }
    } catch (error) {
      showMessage(PAYMENT_FAILED);
      return error.response;
    }
  }

  return (
    <div className={(CN)}>
      <StripeCheckout
        stripeKey={stripeKey}
        token={handleToken}
        amount={totalCount * 100}
        billingAddress
        shippingAddress
      />
    </div>
  );
};

export default Payment;
