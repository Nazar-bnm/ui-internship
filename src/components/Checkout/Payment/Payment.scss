@import '../../../scss/global_styles';

.order-review {
  font-family: font(primary-light);

}
