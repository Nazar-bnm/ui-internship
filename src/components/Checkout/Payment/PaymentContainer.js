import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import showMessage from '../../../actions/actionsNotifications';
import Payment from './Payment';

const mapStateToProps = ({ reducerCart, productsReducer }) => ({
  userCart: reducerCart.userCart,
  products: productsReducer.products,
  totalCount: reducerCart.totalCount
});

const mapDispatchToProps = (dispatch) => ({
  showMessage: bindActionCreators(showMessage, dispatch)
});
export default connect(mapStateToProps, mapDispatchToProps)(Payment);
