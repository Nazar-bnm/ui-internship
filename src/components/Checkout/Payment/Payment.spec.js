import React from 'react';
import { shallow } from 'enzyme';

import Payment from './Payment';

describe('<OrderPreview />', () => {
  let wrapper = shallow(<Payment totalCount={10} showMessage={() => {}} />);
  
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
