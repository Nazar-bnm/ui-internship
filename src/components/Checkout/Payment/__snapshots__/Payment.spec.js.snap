// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderPreview /> should match snapshot 1`] = `
<div
  className="payment"
>
  <ReactStripeCheckout
    ComponentClass="span"
    amount={1000}
    billingAddress={true}
    className="StripeCheckout"
    label="Pay With Card"
    locale="auto"
    reconfigureOnUpdate={false}
    shippingAddress={true}
    stripeKey="pk_test_q7q6RtscsNRaRZnaBHuoRC7x00fupiksw6"
    token={[Function]}
    triggerEvent="onClick"
  />
</div>
`;
